examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -738,34 +770,32 @@ func (g *genDeepCopy) doInterface(t *types.Type, sw *generator.SnippetWriter) {
 }
 
 func (g *genDeepCopy) doPointer(t *types.Type, sw *generator.SnippetWriter) {
+	ut := underlyingType(t)
+	uet := underlyingType(ut.Elem)
+
 	sw.Do("if *in == nil { *out = nil } else {\n", t)
-	if hasDeepCopyMethod(t.Elem) {
-		sw.Do("*out = new($.Elem|raw$)\n", t)
+	if hasDeepCopyMethod(ut.Elem) {
```

Comment:
I think it's worth the comment you left me [here](https://github.com/kubernetes/gengo/pull/73#issuecomment-354788061) why we check `ut.Elem` but generate for `uet`. It's something that needs to be cleared to everyone reading the code.

Reply:
done